src/filesys.cpp

```diff
@@ -142,7 +142,7 @@ bool RecursiveDelete(const std::string &path)
 		infostream<<"RecursiveDelete: Deleting content of directory "
 				<<path<<std::endl;
 		std::vector<DirListNode> content = GetDirListing(path);
-		for(int i=0; i<content.size(); i++){
+		for(unsigned int i=0; i<content.size(); i++){
```

Comment:
This should be `size_t` unsigned int can possibly overflow, and create an endless loop (or a crash when it tries to delete something it already has deleted). Also, you should fix the spaces, so that it looks like:
`for (size_t i = 0; i < content.size(); i++) {`